### PR TITLE
DOC-998: Make Facebook App credentials more detailed

### PR DESCRIPTION
**Pull request by freakwriter**
Summary of changes:

- Actually figured out how to get the credential to work, added very detailed steps for that.
- Updated Facebook Trigger at the same time since I was already in here testing everything. Major changes there include: separate subpages for each Object, related resources for them, prerequisites when it appeared that the webhook required more configuration.

### CODE REVIEW

**Comment by freakwriter** [comment]
@nik8n I finally got this credential working. This branch handles both the credential as well as updating the Facebook trigger docs, since I got sucked into that rabbithole as I tested credentials anyway.

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/application.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Application** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/application.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```
Maybe again

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/certificate-transparency.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Certificate Transparency** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/certificate-transparency.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/group.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Group** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/group.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/instagram.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Instagram** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/instagram.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/link.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Link** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/link.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    ```
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Page** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include individual profile fields, as well as:
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md:40]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include individual profile fields, as well as:
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    * **Feed**: Describes most changes to a page's feed, including posts, likes, shares, and so on.
+    * **Leadgen**: Notifies you when a page's lead generation settings change.
+    * **Live Videos**: Notifies you when a page's live video status changes.
+    * **Mention**: Notifies you when new mentions in pages, comments, and so on occur.
+    * **Merchant Review**: Notifies you when a page's merchant review settings change.
+    * **Page Change Proposal**: Notifies you when Facebook suggests proposed changes for your Facebook Page.
+    * **Page Upcoming Change**: Notifies you when about upcoming changes that will occur on your Facebook Page. Facebook has suggested these changes and they may have a deadline to accept or reject before automatically taking effect.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md:47]:
```suggestion
    * **Page Upcoming Change**: Notifies you about upcoming changes that will occur on your Facebook Page. Facebook has suggested these changes and they may have a deadline to accept or reject before automatically taking effect.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    * **Product Review**: Notifies you when a page's product review settings change.
+    * **Ratings**: Notifies you when a page's ratings change, including new ratings or a user comments on or reacts to a rating.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/page.md:49]:
```suggestion
    * **Ratings**: Notifies you when a page's ratings change, including new ratings or when a user comments on or reacts to a rating.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/permissions.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+#https://www.notion.so/n8n/Frontmatter-432c2b8dff1f43d4b1c8d20075510fe4
+title: Facebook trigger Permissions Object
+description: Documentation for the Permissions Object of the Facebook trigger node in n8n, a workflow automation platform. Includes configuration details.
+contentType: integration
+priority: medium
+---
+
+# Permissions (Facebook trigger)
+
+Use this object to receive updates when a user grants or revokes a permission to your app. Refer to [Facebook trigger](/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/) for more information on the trigger itself.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/permissions.md:11]:
```suggestion
Use this object to receive updates when a user grants or revokes a permission for your app. Refer to [Facebook trigger](/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/) for more information on the trigger itself.
```
Might read a bit better with "revokes".

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/permissions.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Permissions** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/permissions.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/user.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **User** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/user.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/whatsapp.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+1. If you are working as a Solution Partner, make sure your app has completed App Review and requested the `whatsapp_business_management` permission.
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **WhatsApp Business Account** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/whatsapp.md:39]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
```

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/workplace-security.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+///
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Workplace Security** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```
Comment (imchairmanm) [inline docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/workplace-security.md:28]:
```suggestion
1. **Field Names or IDs**: By default, the node will trigger on all the available events using the `*` wildcard filter. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in.
```

**Review by imchairmanm — APPROVED** [review]
Feel free to ignore or change all of the wildcard comments if you think they're unnecessary or confusing.

File: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/ad-account.md [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+## Trigger configuration
+
+To configure the trigger with this Object:
+
+1. Select the **Credential to connect with**. Select an existing or create a new [Facebook App credential](/integrations/builtin/credentials/facebookapp/).
+1. Enter the **APP ID** of the app connected to your credential. Refer to the [Facebook App credential](/integrations/builtin/credentials/facebookapp/) documentation for more information.
+1. Select **Ad Account** as the **Object**.
+1. **Field Names or IDs**: By default, the node will trigger on all the available Ad Account events using the `*`. If you'd like to limit the events, use the `X` to remove the star and use the dropdown or an expression to select the updates you're interested in. Options include:
+    * **In Process Ad Objects**: Notifies you when a campaign, ad set, or ad exits the `IN_PROCESS` status. See post processing for more information.
```
[the remark this answers is not in the document]
Comment (freakwriter) [reply docs/integrations/builtin/trigger-nodes/n8n-nodes-base.facebooktrigger/ad-account.md]: Good catch. That was a copy/paste fail I overlooked from the Meta docs. Updated.

**Pull request merged**